pkg/drawio/styles.go

```diff
@@ -189,20 +208,31 @@ func (stl *templateStyles) setCanTypeHaveAMiniIcon(nodes []TreeNodeInterface) {
 }
 
 // ////////////////////////////////////////////////////////////////////////////////////////
+func (stl *templateStyles) representingType(tn TreeNodeInterface) reflect.Type {
+	if stl.provider != common.AWS || reflect.TypeOf(tn).Elem() != reflect.TypeOf(SubnetTreeNode{}) {
+		return reflect.TypeOf(tn).Elem()
+	}
+	if tn.(*SubnetTreeNode).IsPublic() {
+		return reflect.TypeOf(publicSubnetTreeNode{})
+	}
+	return reflect.TypeOf(privateSubnetTreeNode{})
+}
```

Comment:
This looks somewhat ugly.
Would it make more sense to export `publicSubnetTreeNode` and `privateSubnetTreeNode` as specializations of `SubnetTreeNode` instead of having the public flag?